src/rrt/Tree.hpp

```diff
@@ -309,7 +382,13 @@ class Tree {
         }
 
         // Add a node to the tree for this state
-        _nodes.push_back(Node<T>(intermediateState, source));
+        Node<T> newNode =
+            Node<T>(intermediateState, source, _dimensions, _TToArray);
+        _kdtree.addPoints(
+            flann::Matrix<double>(newNode.coordinates(), 1, _dimensions));
+        _nodes.push_back(newNode);
```

Comment:
This isn't really safe (if the coordinates get deleted correctly). You are making a copy of newNode right now when you put it into the matrix. It would probably be safest to emplace back and then do addpoints. If Node classes shouldn't be copied, you might also want to consider deleting the copy constructor so you don't accidentally do it. If you get rid of the raw pointer, it might be ok to copy it (unless your using .coordinates and then copying).